modules/core/04-channel/keeper/grpc_query_test.go

```diff
@@ -1539,3 +1539,91 @@ func (suite *KeeperTestSuite) TestQueryUpgradeSequence() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestQueryUpgradeError() {
+	var (
+		req        *types.QueryUpgradeErrorRequest
+		upgradeErr types.ErrorReceipt
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"empty request",
+			func() {
+				req = nil
+			},
+			false,
+		},
+		{
+			"invalid port ID",
+			func() {
+				req = &types.QueryUpgradeErrorRequest{
+					PortId:    "",
+					ChannelId: "test-channel-id",
+				}
+			},
+			false,
+		},
+		{
+			"invalid channel ID",
+			func() {
+				req = &types.QueryUpgradeErrorRequest{
+					PortId:    "test-port-id",
+					ChannelId: "",
+				}
+			},
+			false,
+		},
+		{
+			"channel not found",
+			func() {
+				req = &types.QueryUpgradeErrorRequest{
+					PortId:    "test-port-id",
+					ChannelId: "test-channel-id",
+				}
+			},
+			false,
+		},
+		{
+			"success",
+			func() {
+				path := ibctesting.NewPath(suite.chainA, suite.chainB)
+				suite.coordinator.Setup(path)
+				upgradeErr = types.ErrorReceipt{
```

Comment:
mega nit: I think @chatton added a `types.NewErrorReceipt()` function you can use here

Reply:
ah true, i will use this